apps/system/test/unit/statusbar_test.js

```diff
@@ -658,6 +484,12 @@ suite('system/Statusbar', function() {
       assert.isTrue(Statusbar.element.classList.contains('fullscreen-layout'));
       assert.isTrue(stub.calledOnce);
     });
+
+    test('setAppearance maximized with fewIcons', function() {
+      Statusbar._numMinimizedIcons = 5;
+      Statusbar.setAppearance();
+      assert.isFalse(Statusbar.element.classList.contains('maximized'));
+    });
```

Comment:
and to cover the `iconhidden` handler without inspecting the internal data structure we can have
* 1 test just below the limit of icons count, calling `setAppearance` -> maximized not toggled
* 1 test with an `iconshow`, pushing over the limit -> maximized toggled right away without waiting for `setAppearance`
* 1 test with an `iconhidden` after that -> calling `setAppearance` -> maximized is not toggled (because we're back below the limit)